models/org_team.go

```diff
@@ -293,44 +293,44 @@ func UpdateTeam(t *organization.Team, authChanged, includeAllChanged bool) (err
 
 // DeleteTeam deletes given team.
 // It's caller's responsibility to assign organization ID.
-func DeleteTeam(t *organization.Team) error {
-	ctx, committer, err := db.TxContext(db.DefaultContext)
+func DeleteTeam(ctx context.Context, t *organization.Team) error {
+	dbCtx, committer, err := db.TxContext(ctx)
```

Comment:
It should keep using the name `ctx`, otherwise it might cause bugs like:
*  #26684 
* (and the fix #26763)

Reply:
The problem is, we already have a variable called `ctx` on this function now. IF we overwrite this variable, it could cause confusion and bugs e.g. using `ctx `after the `TxContext` is closed.

Reply:
No need to worry about it. It's safe to shadow the parameter `ctx` by a local `ctx`.

Because such functions should always end with `return committer.Commit()`. Even if TxContext is closed, it's still safe to use the `ctx` for other purposes.

Reply:
> Because such functions should always end with return `committer.Commit()`

Not always. Take for example a look at the [AddTeamMember](https://github.com/go-gitea/gitea/blob/941de08557ec32f26992d60d3465ebbe829c038c/models/org_team.go#L359) function from this PR. `ctx` is used after `committer.Close()` is called. Would this also be safe?

Reply:
I think it is a design bug, it is already wrong. For example, the code below shouldn't access `ctx` in goroutine, because the ctx could be canceled if the browser finishes the request.

Reply:
So I should use `db.DefaultContext` inside the goroutine?

Reply:
As a quick fix, the AddTeamMember code should use `db.DefaultContext` when the transaction is closed ...... and leave the complete fix to the future.

Reply:
Then I will do that and replace `dbCtx` with `ctx `in my code and hope nothing goes wrong.

Reply:
Should be done now. I hope I don't missed something.